plugins/analytics/public/censysio.py

```diff
@@ -0,0 +1,135 @@
+import json
+import requests
+from core.observables import Observable
+from core.analytics import OneShotAnalytics
+from core.observables import Ip, Certificate, Text, Hostname, Hash, AutonomousSystem
+from core.config.config import yeti_config
+from core.errors import GenericYetiError, ObservableValidationError
+
+class CensysApi(object):
+    """
+        https://censys.io/api
+    """
+    settings = {
+        "urlscanio_secret": {
+            "name": "censys_secret",
+            "description": "Secret provided by Censys.io."
+        },
+        "urlscanio_api_key": {
+            "name": "censys_apikey",
+            "description": "API Key provided by Censys.io."
+        }
+    }
+
+    API_URL = "https://censys.io/api/v1/view"
+
+    @staticmethod
+    def _process_data(json_result, observable):
+
+        links = set()
+
+        if isinstance(observable, Ip):
+            if json_result.get('autonomous_system'):
+                if json_result['autonomous_system'].get('asn'):
+                    asn = AutonomousSystem.get_or_create(value=str(json_result['autonomous_system']['asn']))
+                    links.update(asn.active_link_to(observable, 'asn#', 'Censys Query'))
+
+                if json_result['autonomous_system'].get('name'):
+                    asnname = Text.get_or_create(value=json_result['autonomous_system']['name'])
+                    links.update(asnname.active_link_to(observable,
+                        'asn_name', 'Censys Query'))
+
+                if json_result['autonomous_system'].get('routed_prefix'):
+                    routed_prefix = Text.get_or_create(value=json_result['autonomous_system']['routed_prefix'])
+                    links.update(routed_prefix.active_link_to(observable,
+                        'routed_prefix', 'Censys Query'))
+
+            json_result = json_result.get("443", {}).get('https', {}).get('tls', {}).get("certificate", {})
```

Comment:
Why do you attempt to get a certificate only if the observable is an IP? This also modifies the json_result you're inspecting in line 49. It's a bit unclear what assumptions you're making about json_result at this point. Please add comments to make this clearer

Reply:
that is bcz, IP has a bit more the data than the rest of iocs,